GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -523,7 +523,27 @@ public bool TryLoad(out string error)
                 this.UpgradeRing = RingType.NoConfig;
 
                 string ringConfig = null;
-                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, out ringConfig, out error))
+                string upgradeFeedUrl = null;
+
+                // 1. Try to read ring from upgrade.feedurl first.
+                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl, out upgradeFeedUrl, out error) &&
+                    !string.IsNullOrEmpty(upgradeFeedUrl))
+                {
+                    string[] expectedRings = new string[] { RingType.Slow.ToString(), RingType.Fast.ToString() };
+                    foreach (string ring in expectedRings)
+                    {
+                        string ringSubstring = $"@{ring}/";
+                        if (upgradeFeedUrl.IndexOf(ringSubstring, StringComparison.OrdinalIgnoreCase) != -1)
+                        {
+                            ringConfig = ring;

```

Comment:
Can some unit test(s) be added to validate the new functionality?

Reply:
There is some discussion about what the behavior should be - I think unit tests would be a great way to clarify the expected behavior under different conditions.

Reply:
Added couple UT to demonstrate 
1. feedURL config would override upgrade.ring config. 
2. Upgrade fails when feedURL config exists but does not contain ring info.